master/buildbot/changes/p4poller.py

```diff
@@ -125,7 +125,7 @@ def __init__(self, p4port=None, p4user=None, p4passwd=None,
                  pollInterval=60 * 10, histmax=None, pollinterval=-2,
                  encoding='utf8', project=None, name=None,
                  use_tickets=False, ticket_login_interval=60 * 60 * 24,
-                 server_tz=None, pollAtLaunch=False, revlink=lambda branch, revision: (u'')):
+                 server_tz=None, pollAtLaunch=False, revlink=lambda branch, revision: (u''), resolvewho=lambda who: (who)):
```

Comment:
Any way this could be done as a renderable instead?

Reply:
@seankelly - I'm not sure I understand how a renderable would be appropriate here. I'm open to any suggestions though. 

In our case we take the 'who' in the form of user@workspace - for example Charles@SecondWorkspace - and need to map it to charles.oneal@workspace.com. We pipe it through a callable that has LDAP and perforce user list knowledge, so it is a couple of dictionary checks and some additional logic. How do renderables work in this case?

Reply:
A renderable can be a function to do that. It meshes better with how Buildbot handles dynamic text. The issue is that its input is the build's properties.

I'm hoping @tardyp might have an idea.

Reply:
yeah, I've thought of that too, but renderable really do not have anything to interpolate in pollers